src/crimson/common/gated.h

```diff
@@ -31,13 +31,13 @@ class Gated {
 
   template <typename Func, typename T>
   inline void dispatch_in_background(const char* what, T& who, Func&& func) {
-    ceph_assert(seastar::this_shard_id() == sid);
+    //ceph_assert(seastar::this_shard_id() == sid);
```

Comment:
If this assert fails in other cases, can you take the chance to point out which user's gate is/might-be related?

Reply:
@cyx1231st its getting out of the scope of that PR, I opened new tracker and will work on it as well